crates/transaction-pool/src/lib.rs

```diff
@@ -473,6 +473,14 @@ where
         self.pool.get_transactions_by_sender(sender)
     }
 
+    fn get_transactions_by_sender_and_nonce(
+        &self,
+        sender: Address,
+        nonce: u64,
+    ) -> Option<Arc<ValidPoolTransaction<Self::Transaction>>> {
+        self.pool.get_transactions_by_sender(sender).into_iter().find(|tx| tx.nonce() == nonce)
```

Comment:
let's go all the way and impl this one layer deeper so we can get rid of the vec alloc, this is a single `all_transactions.all` map lookup with the TransactionId (Sender, nonce)

Reply:
understood, just pushed a fix, tell me if this does the job

Reply:
or should we use 

```rust
self.pooled_transactions().iter().find(|tx| tx.sender() == sender && tx.nonce() == nonce)
```